src/bitcoin/blockhash.h

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2019 The Bitcoin developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_PRIMITIVES_BLOCKHASH_H
+#define BITCOIN_PRIMITIVES_BLOCKHASH_H
+
+#include "uint256.h"
+
+namespace bitcoin{
+/**
+ * A BlockHash is a unqiue identifier for a block.
+ */
+struct BlockHash : public uint256 {
+    explicit BlockHash() : uint256() {}
+    explicit BlockHash(const uint256 &b) : uint256(b) {}
```

Comment:
Why `explicit` default c'tor?!?

```suggestion
    constexpr BlockHash() noexcept = default;
    explicit constexpr BlockHash(const uint256 &b) noexcept : uint256(b) {}
```